deploy/charts/mixer/templates/deployment.yaml

```diff
@@ -0,0 +1,166 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+{{- range $index, $serviceGroup := .Values.serviceGroups }}
+{{- range $serviceName, $unused := . }}
+{{- if gt $index 0 }}
+---
+{{- end  }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "mixer.fullname" $ }}-{{ $serviceName }}
+  namespace: {{ $.Values.namespace.name }}
+  labels:
+    service:  {{ include "mixer.fullname" $ }}-{{ $serviceName }}
+    {{- include "mixer.labels" $ | nindent 4 }}
+spec:
+  replicas: {{ .replicas.default }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      # maximum number of Pods that can be created over the desired number of Pods
+      maxSurge: 4
+      # Maximum number of pods that can be unavailable during the update process
+      # Environment have different number of pods, this is roughly 8% in prod.
+      maxUnavailable: 5
```

Comment:
may need to update maxSurge and maxUnavailable as some services could have small number of replicas now?

Reply:
Missed this part, thanks!  Not sure what the best values are so for the time being both max surge and unavailable is 25% of default + 1 pod.